File: DFS.java
import java.util.*;
import java.io.*;

class Graph
{
	int V;
	LinkedList<Integer> stack[];

	Graph(int v)
	{
		V=v;
		stack=new LinkedList[v];
		for(int i=0;i<v;i++)
		{
			stack[i]=new LinkedList<Integer>();
		}
	}
	void addList(int u,int v)
	{
		stack[u].add(v);
	}

	void UtilDfs(int s,boolean vis[])
	{
		vis[s]=true;
		System.out.print(s+" ");

		Iterator<Integer> it=stack[s].iterator();
		while(it.hasNext())
		{
			int n=it.next();
			if(!vis[n])
			{
				vis[n]=true;
				UtilDfs(n,vis);
			}
		}

	}
	
	void Dfs(int s)
	{
		boolean[] vis=new boolean[V];
		UtilDfs(s,vis);
	}

	public static void main(String[] args)
	{
		Scanner sc=new  Scanner(System.in);
		int e,v;
		
		e=sc.nextInt();
		v=sc.nextInt();

		//boolean[] vis=new boolean[v];
		Graph g=new Graph(v);
		for(int i=0;i<e;i++)
		{
			int v1,v2;
			v1=sc.nextInt();
			v2=sc.nextInt();

			g.addList(v1,v2);
		}
		g.Dfs(0);

	}
}

File: Dfs.py
from collections import defaultdict

class Graph:
    
    def __init__(self):
        self.graph=defaultdict(list)
        
    def edgeList(self,v,u):
        self.graph[v].append(u)
    
    def UtilDfs(self,src,vis):
        vis[src]=True 
        
        print(src,end=",")
        
        for i in self.graph[src]:
            if vis[i]==False:
                vis[i]=True
                self.UtilDfs(i,vis)
    
    def Dfs(self,x):
        
        
        v=len(self.graph)

        visited=[False]*(V+1) 
        self.UtilDfs(x,visited)
        
               
e,V=map(int,input().split())

g=Graph()
for i in range(e):
    v1,v2=map(int,input().split())
    if(i==0):
        x=v1 
    g.edgeList(v1,v2)
g.Dfs(x)
